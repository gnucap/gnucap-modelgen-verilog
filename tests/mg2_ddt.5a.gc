attach ../modelgen_0.so
attach ../d_vasrc.so

* BUG
* options noincmode

spice
.subckt isin p n
.parameter ampl
.parameter dc
I1 p n sin offset=-1 freq=1 amplitude={ampl} delay=-.25 dc=dc
.ends

.verilog

`modelgen
module test_ddt5a(p, n, x);
	(* desc="" *) real d0;
	electrical p, n, x;
	analog begin
		d0 = ddt(I(p,n));
		V(x) <+ d0;
	end
endmodule

!make test_ddt5a.so > /dev/null
attach ./test_ddt5a.so

verilog
parameter r=1
parameter v=0

test_ddt5a #() dut(1,0,x);
isin #(.dc(v), .ampl(1)) i1(0, 1);
// resistor #(1.) r(0,1);

list

print tran i(i1.*) v(nodes) d0(dut) d1(dut) iter(0)
tran 1  trace=n
status notime
end
