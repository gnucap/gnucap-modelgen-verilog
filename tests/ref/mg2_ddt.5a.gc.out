module isin (p,n);
  .parameter ampl 
  .parameter dc 
  isource #(sin.offset(-1.), .amplitude(ampl), .frequency( 1.), .delay(-0.25), .damping(NA( 0.))  DC dc) I1 (.p(p),.n(n));
endmodule // isin

parameter r=1 
parameter v=0 
test_ddt5a #() dut (.p(1),.n(0),.x(x));
isin #(.ampl(1),.dc(v)) i1 (.p(0),.n(1));
// resistor #(1.) r(0,1); 
#Time       i(i1.I1)   v(1)       v(x)       d0(dut)    d1(dut)    iter(0)   
 0.         0.         0.         0.         0.         ??         3.        
 80.376n   -127.56f    0.        -1.5871u   -1.5871u    ??         18.       
 212.69n   -892.95f    0.        -9.982u    -9.982u     ??         6.        
 345.01n   -2.3496p    0.        -12.035u   -12.035u    ??         3.        
 2.356u    -109.56p   -1.f       -94.595u   -94.595u    ??         3.        
 15.407u   -4.6856n   -47.f      -606.66u   -606.66u    ??         3.        
 570.03u   -6.4139u   -64.139p   -0.022505  -0.022505   ??         3.        
 0.0092607 -0.0016924 -16.924n   -0.36549   -0.36549    ??         3.        
 0.042285  -0.035088  -350.88n   -1.657     -1.657      ??         3.        
 0.092691  -0.16485   -1.6485u   -3.4919    -3.4919     ??         3.        
 0.1431    -0.37769   -3.7769u   -4.9531    -4.9531     ??         3.        
 0.1935    -0.65243   -6.5243u   -5.948     -5.948      ??         3.        
 0.24391   -0.96174   -9.6174u   -6.3248    -6.3248     ??         3.        
 0.29432   -1.2749    -12.749u   -6.099     -6.099      ??         3.        
 0.36488   -1.6608    -16.608u   -4.8381    -4.8381     ??         3.        
 0.43545   -1.9189    -19.189u   -2.4772    -2.4772     ??         3.        
 0.50602   -1.9993    -19.993u    0.19842    0.19842    ??         3.        
 0.57659   -1.8864    -18.864u    3.0001     3.0001     ??         3.        
 0.71773   -1.2014    -12.014u    6.7073     6.7073     ??         3.        
 0.85886   -0.3681    -3.681u     5.1011     5.1011     ??         3.        
 0.92943   -0.096699  -966.99n    2.5907     2.5907     ??         3.        
 0.96472   -0.024474  -244.74n    1.5032     1.5032     ??         3.        
 1.         0.         0.        -0.11598   -0.11598    ??         3.        
Gnucap   System status
iterations: op=0, dc=0, tran=87, fourier=0, total=116
transient timesteps: accepted=23, rejected=6, total=29
nodes: user=2, subckt=0, model=1, total=3
dctran density=100.0%, ac density=100.0%
